src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSslVersion.cs

```diff
@@ -12,16 +12,28 @@ internal static partial class OpenSsl
         private static Version s_opensslVersion;
 
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_OpenSslVersionNumber")]
-        internal static extern uint OpenSslVersionNumber();
+        internal static extern long OpenSslVersionNumber();
 
         internal static Version OpenSslVersion
         {
             get
             {
                 if (s_opensslVersion == null)
                 {
-                    uint versionNumber = OpenSslVersionNumber();
-                    s_opensslVersion = new Version((int)(versionNumber >> 28), (int)((versionNumber >> 20) & 0xff), (int)((versionNumber >> 12) & 0xff));
+                    // OpenSSL version numbers are encoded as
+                    // 0xMNNFFPPS: major (one nybble), minor (one byte, unaligned),
+                    // "fix" (one byte, unaligned), patch (one byte, unaligned), status (one nybble)
+                    //
+                    // e.g. 1.0.2a final is 0x1000201F
+                    //
+                    // Currently they don't exceed 29-bit values, but we use long here to account
+                    // for the expanded range on their 64-bit C-long return value.
+                    long versionNumber = OpenSslVersionNumber();
+                    int major = (int)((versionNumber >> 28) & 0xF);
+                    int minor = (int)((versionNumber >> 20) & 0xFF);
+                    int fix = (int)((versionNumber >> 12) & 0xFF);
+
+                    s_opensslVersion = new Version(major, minor, fix);
```

Comment:
Where is this used?  I'm just surprised to see us parsing it like this, given that we explicitly opted to expose a long due to it being opaque data.  Is this just for test code?  And if so, should this OpenSslVersion property be moved into a test assembly somewhere?

Reply:
>  Is this just for test code?

Yeah.

>  And if so, should this OpenSslVersion property be moved into a test assembly somewhere?

What a remarkably reasonable question that means I have more work to do.  Okie dokie :smile:.